src/v2/QueryExecutor.sol

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: MIT
```

Comment:
99% of this contract is a carbon copy from the `QueryManager` in v1. We are moving it to a standalone contract for better modularization.